Add validation for capability value - updating the value

AnishReddyRavula opened the pull request:
Earlier only the validation for capability value - if the value for the capability already exists is checked when the capability is first assigned to the node.

Does not consider the case, when the value is updated for a node. Adding another method in models to check if any node has the same value for a capability when is_unique is True

Change-Id: I115a6de12cabc2e32b5c10669b0f9c7b0fdbb5f5

AnishReddyRavula committed:
Add validation for capability value - updating the value

Earlier only the validation for capability value - if the value for the capability
already exists is checked when the capability is first assigned to the node.

Does not consider the case, when the value is updated for a node.
Adding another method in models to check if any node has the same value for a capability
when is_unique is True

Change-Id: I115a6de12cabc2e32b5c10669b0f9c7b0fdbb5f5

Mark-Powers reviewed (changes requested):
This seems mostly copy/paste, can it be refactored into a helper method?

AnishReddyRavula committed:
improved validation in computehost extra capability model

Previously, validation was only performed for the `capability_id` and `capability_value` attributes separately.
This update consolidates the validation logic into a single method called `validate_capability`, which is invoked for both attributes.
Additionally, the docstring for the method has been updated to provide clarity on when each validation scenario occurs.

Change-Id: I149784b5e4be71b34dfebad470cc645fed5656b4
